ledger_device_sdk/src/io.rs

```diff
@@ -331,6 +331,44 @@ impl Comm {
                 return None;
             }
 
+            // Default BOLOS APDU Handling
+            let apdu_header = self.get_apdu_metadata();
```

Comment:
Maybe put the whole handling logic in a private function ? It would keep things tighter in the main decode function.